setup.sh

```diff
@@ -104,3 +104,16 @@ echo "${RUNNER_TOOL_CACHE}/dart-sdk/bin" >> $GITHUB_PATH
 # Report success, and print version.
 echo -e "Successfully installed Dart SDK:"
 ${RUNNER_TOOL_CACHE}/dart-sdk/bin/dart --version
+
+# When enabled through env variables, create OIDC token for publishing on pub.dev.
+if [[ "${ACTIONS_ID_TOKEN_REQUEST_URL}" != "" && "${ACTIONS_ID_TOKEN_REQUEST_TOKEN}" != "" ]]
+then
+  if [[ -x "$(command -v jq)" ]]
+  then
+    PUB_TOKEN=$(curl --retry 5 --retry-connrefused -sLS "${ACTIONS_ID_TOKEN_REQUEST_URL}&audience=https://pub.dev" -H "User-Agent: actions/oidc-client" -H "Authorization: Bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" | jq -r '.value')
+    echo "PUB_TOKEN=${PUB_TOKEN}" >> $GITHUB_ENV
+    ${RUNNER_TOOL_CACHE}/dart-sdk/bin/dart pub token add https://pub.dev --env-var PUB_TOKEN
```

Comment:
To avoid the line: `Environment variable "PUB_TOKEN" is not defined.` in logs, we should define `PUB_TOKEN` when calling `dart pub token add` -- otherwise it'll warn us that it's not defined.